docs/advanced-guides/oauth-client.md

```diff
@@ -0,0 +1,293 @@
+
+The [atproto OAuth specification](https://github.com/bluesky-social/atproto-website/pull/326) is the authoritative document on how to use OAuth with atproto. If there are discrepancies between this document and the specification, defer to the specification. This document skips over a few details and uses more atproto-specific terminology, but readers are still expected to be familiar with OAuth 2 concepts, terminology, and standards.
+
+This guide is focused on apps which use OAuth to make authorized ("authz") requests to user PDS servers, for example to write records to atproto repositories, or make proxied API requests to other services. It is also possible to use the atproto identity system only for authentication ("authn"), similar to OpenID/OIDC.
+
+## Types of Clients
+
+This guide covers three simplified types of OAuth client:
+
+- **Web Services**: traditional web apps which involve a server/backend running code to make actual PDS requests and talk with a database. There is some form of auth between browsers and the web service, such as cookie sessions; this auth layer is distinct from OAuth. The server may return complete HTML pages, or there may be an API between code running in the browser and code running on the server. Integrations with and extensions of existing web services also fall under this category.
+- **Browser Apps**: "single-page" applications which run in a web browser, implemented using web platform APIs and JavaScript or WASM runtimes. The server-side ("backend") component is minimal, or even just static file hosting.
+- **Mobile and Desktop Apps**: what they sound like: "native" apps that run on mobile operating systems (smartphones, tablets, etc), or desktop applications
+
+|                         | **Web Service** | **Browser App** | **Mobile or Desktop App** |
+| ---                     | ---             | ---             | ---                       |
+| **OAuth 2 Client Type** | "Confidential"  | "Public"        | "Public"                  |
+| `client_id` | ✅ URL to metadata | ✅ URL to metadata | ✅ URL to metadata |
+| `client_secret` | ❌  | ❌  | ❌  |
+| **OAuth 2 Grant Types** | `authorization_code`
+`refresh_token` | `authorization_code`
+`refresh_token` | `authorization_code`
+`refresh_token` |
+| **Client Metadata** | ✅ Public Web | ✅ Public Web | ✅ Public Web |
+| **Client Metadata JWK** | ✅ Public Web | ❌  | ❌  |
+| **PKCE** | ✅ | ✅ | ✅ |
+| **PAR** | ✅ | ✅ | ✅ |
+| **DPoP** | ✅ | ✅ | ✅ |
+| **Handle Resolution** | DNS and HTTPS | via helper service | DNS and HTTPS or via helper service |
```

Comment:
"Helper service" could be DNS-over-HTTP, which is something we actually offer an adapter for in our SDK.